tests/Mooc/Courses/Application/Find/CourseFinderTest.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CodelyTv\Tests\Mooc\Courses\Application\Find;
+
+use CodelyTv\Mooc\Courses\Application\Find\CourseFinder;
+use CodelyTv\Mooc\Courses\Domain\Course;
+use CodelyTv\Mooc\Courses\Domain\CourseNotExist;
+use CodelyTv\Tests\Mooc\Courses\CoursesModuleUnitTestCase;
+use CodelyTv\Tests\Mooc\Courses\Domain\CourseMother;
+
+final class CourseFinderTest extends CoursesModuleUnitTestCase
+{
+    /**
+     * @var CourseFinder
+     */
+    private $finder;
+
+    /**
+     * @var Course
+     */
+    private $course;
+
+    /**
+     * @var Course|null
+     */
+    private $actualResult;
+
+    public function testFindCourse()
+    {
+        $this->givenSubjectUnderTest();
+        $this->givenAnExistingCourse();
+        $this->shouldSearch($this->course->id(), $this->course);
+
+        $this->whenISearchForTheExistingCourse();
+
+        $this->thenTheCourseIsReturned();
```

Comment:
This could be done like
```suggestion
        $this->whenISearchForTheCourse();
        $this->theCourseShouldBeSearched();

        $this->run();
```
But to be able to do that, you should do some test refactor 😬